python/tvm/relay/frontend/tensorflow.py

```diff
@@ -574,25 +590,62 @@ def _impl(inputs, attr, params):
         except AttributeError:
             # Shape operator is already pruned, hence
             # try to infer shape by precompute prune if possible.
-            func = _expr.Function(ir_pass.free_vars(inputs[1]), inputs[1])
-            with tvm.relay.build_config(opt_level=0):
-                graph, lib, params = tvm.relay.build(func, target="llvm", params=params)
-            ctx = tvm.context("llvm", 0)
-            from tvm.contrib import graph_runtime
-            m = graph_runtime.create(graph, lib, ctx)
-            m.set_input(**params)
-            m.run()
-            params_new = m.get_output(0)
+            params_new = _infer_value(inputs[1], params)
             inputs.pop(1)
             return AttrCvt(
                 op_name="reshape",
                 extras={'newshape':tuple(params_new.asnumpy().astype('int64').flatten())},
                 ignores=['Tshape'])(inputs, attr)
     return _impl
 
+
+# Currently only supports NHWC, TODO add NCHW support.
```

Comment:
It looks you handled HCHW below, is this TODO needed?

Reply:
Whoops, you're right. Removed that comment in latest push.